src/Effects.js

```diff
@@ -1445,6 +1445,40 @@ module.exports = class Effects {
         }
       }
     };
+
+    this.music_wave = {
+      wave: [],
```

Comment:
It looks like we'll only ever have one wave - would it make sense to update this to be a single variable and not an array?
You should be able to do that my replacing `wave: []` with `wave: null` or an empty object and then updating lines 1459 and 1467 to assign and reference `this.wave` as an object.

Reply:
I made wave an object.